cmd/collect.go

```diff
@@ -30,82 +26,75 @@ var collectCmd = &cobra.Command{
 		"  magellan collect --cache ./assets.db --output ./logs --timeout 30 --cacert cecert.pem\n" +
 		"  magellan collect --host smd.example.com --port 27779 --username username --password password",
 	Run: func(cmd *cobra.Command, args []string) {
-		// make application logger
-		l := log.NewLogger(logrus.New(), logrus.DebugLevel)
-
 		// get probe states stored in db from scan
-		probeStates, err := sqlite.GetProbeResults(cachePath)
+		scannedResults, err := sqlite.GetScannedAssets(cachePath)
 		if err != nil {
-			l.Log.Errorf("failed toget states: %v", err)
+			log.Error().Err(err).Msgf("failed to get scanned results from cache")
 		}
 
+		// URL sanitanization for host argument
+		host = strings.TrimSuffix(host, "/")
+		host = strings.ReplaceAll(host, "//", "/")
```

Comment:
This will cause issues in the protocol section of the URL. I.e. it will replace `https://` with `https:/`, which is problematic.

Reply:
Should be fixed now.

Reply:
This was moved to the internal `url` package with better processing, so I'll mark this as resolved.